### PR TITLE
withDockerContainer and sha1 utility steps

### PR DESCRIPTION
**Pull request by HarnessAryan**
added a withDockerContainer and its related file
added a sha1 utility step and its related file
fmt the code

### CODE REVIEW

File: convert/jenkinsjson/covert.go [hunk 1]
```diff
@@ -172,12 +178,12 @@ func collectStagesWithID(jsonNode *jenkinsjson.Node, processedTools *ProcessedTo
...
 
-			stageID := searchChildNodesForStageId(&childNode, stageName)
+			stepId := searchChildNodesForStageId(&childNode, stageName)
 
 			stepsInStage := make([]*harness.Step, 0)
-			recursiveParseJsonToSteps(childNode, &stepsInStage, processedTools, variables)
+			recursiveParseJsonToSteps(childNode, &stepsInStage, processedTools, variables, dockerImage)
 
-			// Create the stepGroup for the new stage
+			// Create the harness stage for each Jenkins stage
```
Comment (jamie-harness) [inline convert/jenkinsjson/covert.go:186]: Can you clean up these changes?

File: convert/jenkinsjson/covert.go [hunk 2]
```diff
@@ -281,28 +286,15 @@ func recursiveParseJsonToSteps(currentNode jenkinsjson.Node, steps *[]*harness.S
 	return nil, nil
 }
 
-func collectStepsWithID(currentNode jenkinsjson.Node, stepWithIDList *[]StepWithID, processedTools *ProcessedTools, variables map[string]string, timeout string) (*harness.CloneStage, *harness.Repository) {
+func collectStepsWithID(currentNode jenkinsjson.Node, stepWithIDList *[]StepWithID, processedTools *ProcessedTools, variables map[string]string, timeout string, dockerImage string) (*harness.CloneStage, *harness.Repository) {
 	var clone *harness.CloneStage
 	var repo *harness.Repository
 
-	if len(currentNode.ParameterMap) == 0 {
```
Comment (jamie-harness) [inline convert/jenkinsjson/covert.go:288]: Can you pull latest

File: convert/jenkinsjson/covert.go [hunk 3]
```diff
@@ -315,61 +307,59 @@ func collectStepsWithID(currentNode jenkinsjson.Node, stepWithIDList *[]StepWith
 			fmt.Println("Error converting step ID to integer:", err)
 		}
 	}
-
+	// fmt.Println("----",currentNode.AttributesMap["jenkins.pipeline.step.type"])
```
Comment (jamie-harness) [inline convert/jenkinsjson/covert.go:310]: Remove comment please

File: convert/jenkinsjson/covert.go [hunk 4]
```diff
@@ -315,61 +307,59 @@ func collectStepsWithID(currentNode jenkinsjson.Node, stepWithIDList *[]StepWith
...
+			clone, repo = collectStepsWithID(child, stepWithIDList, processedTools, variables, timeout, dockerImage)
 		}
-
 	case "stage":
 		// this is technically a step group, we treat it as just steps for now
 		if len(currentNode.Children) > 1 {
 			// handle parallel from parent
 			if currentNode.Children[0].AttributesMap["jenkins.pipeline.step.type"] == "parallel" {
-				parallelStepItemsWithID := make([]StepWithID, 0)
+				parallelStepItems := make([]*harness.Step, 0)
```
Comment (jamie-harness) [inline convert/jenkinsjson/covert.go:338]: ditto

**Review by jamie-harness — CHANGES_REQUESTED** [review]

File: convert/jenkinsjson/covert.go (hunk 2 above)
Comment (HarnessAryan) [reply convert/jenkinsjson/covert.go:288]: done

File: convert/jenkinsjson/covert.go (hunk 3 above)
Comment (HarnessAryan) [reply convert/jenkinsjson/covert.go:310]: done

File: convert/jenkinsjson/covert.go (hunk 2 above)
Comment (jamie-harness) [reply convert/jenkinsjson/covert.go:288]: still not correct it seems

Comment (jamie-harness) [reply convert/jenkinsjson/covert.go:288]: you need to pull across branches

File: convert/jenkinsjson/covert.go (hunk 1 above)
Comment (HarnessAryan) [reply convert/jenkinsjson/covert.go]: yup fixed now

File: convert/jenkinsjson/covert.go (hunk 2 above)
Comment (HarnessAryan) [reply convert/jenkinsjson/covert.go:288]: fixed now

File: convert/jenkinsjson/covert.go (hunk 4 above)
Comment (HarnessAryan) [reply convert/jenkinsjson/covert.go]: also fixed

**Comment by HarnessAryan** [comment]
I have made all the changes as per latest pull now

File: convert/jenkinsjson/covert.go (hunk 2 above)
Comment (jamie-harness) [reply convert/jenkinsjson/covert.go:288]: still not fixed. Maybe you need to do it manually

**Pull request closed without merging**